### ISSUE TITLE
Automatic sync of security group when ports are added not working

### ISSUE THREAD

**Issue by arvindn05** [opensecuritycontroller/osc-core#462]
### Description
Automatic sync of security group with subnet as a member not working.
Automatic sync of security group with VM as a member not working.
### Expected Behavior
When a new VM is added to the subnet that is being protected by the security group, we should automatically sync and protect the new workload.

When a VM is protected and a port is added, automatic sync should work. This is not functional either.

### Actual Behavior
Automatic sync of security group is not working. Manual sync is needs to protect the new workload

### Steps to Reproduce

1. Create a security group with subnet as a member
2. Create a new VM on the subnet
3. Sync of security group is not triggered

1. Create security group with VM as a member
2. Attach an interface to the VM
3. Sync of security group is not triggered

### Additional Information
below Notification is received on OSC. Not sure if the notification is different in Newton/Ocata.
```
2017-09-06 19:18:57,139 [INFO | pool-180-thread-1| OsPortNotificationListener]:  [Port] : message received - {
	"_context_domain": null,
	"_context_roles": ["admin"],
	"event_type": "port.create.end",
	"_context_request_id": "req-413c2547-f91d-4f1a-88ca-2bc2f2a069cc",
	"timestamp": "2017-09-07 02:19:08.292187",
	"_context_tenant_id": "462e190ac1094a7da260e509b8113209",
	"_context_user": "a3f42609d0ba4df5855ea7295263d6e6",
	"_unique_id": "754d2a3a08d647dcb5b029b5fc1ae9c1",
	"_context_resource_uuid": null,
	"_context_tenant_name": "demo",
	"_context_is_admin_project": true,
	"_context_user_id": "a3f42609d0ba4df5855ea7295263d6e6",
	"payload": {
		"port": {
			"status": "DOWN",
			"binding:host_id": "",
			"description": "",
			"allowed_address_pairs": [],
			"tags": [],
			"extra_dhcp_opts": [],
			"updated_at": "2017-09-07T02:19:08Z",
			"device_owner": "",
			"revision_number": 4,
			"binding:profile": {
				
			},
			"fixed_ips": [{
				"subnet_id": "3c5a0907-ff52-4972-9553-e8e6689fd79c",
				"ip_address": "192.168.0.6"
			}],
			"id": "fb4aa263-48ee-4f86-84c3-77fd0ded7a2d",
			"security_groups": ["4c11e50f-5c1a-4080-b7fd-6ad7395db280"],
			"device_id": "",
			"name": "",
			"admin_state_up": true,
			"network_id": "614b5340-d0c0-408e-8b02-9879269bef5a",
			"tenant_id": "462e190ac1094a7da260e509b8113209",
			"binding:vif_details": {
				
			},
			"binding:vnic_type": "normal",
			"binding:vif_type": "unbound",
			"mac_address": "fa:16:3e:25:c9:e5",
			"project_id": "462e190ac1094a7da260e509b8113209",
			"created_at": "2017-09-07T02:19:07Z"
		}
	},
	"_context_project_name": "demo",
	"_context_user_identity": "a3f42609d0ba4df5855ea7295263d6e6 462e190ac1094a7da260e509b8113209 - - -",
	"_context_auth_token": "gAAAAABZsKvOrc_Vl-HhmHudmPTux_9yqwhF8Fu9Tlgn3kdvkJZD_OyzBVZ37D0P1uhPyndBotPKm6ZJ7s5n3KKSTvdY-ldbtI4Q4xMQAYxD71H6nYS5K6jHItx4nAoCNEpX0wa0a_gmNPcddj-mk4e4BCwpxqUSCxiQZaOVxwrnNXIlFJR2QZ-vosN0ILufO5TzuNgDW181",
	"_context_show_deleted": false,
	"_context_tenant": "462e190ac1094a7da260e509b8113209",
	"priority": "INFO",
	"_context_read_only": false,
	"_context_is_admin": true,
	"_context_project_id": "462e190ac1094a7da260e509b8113209",
	"_context_project_domain": null,
	"_context_timestamp": "2017-09-07 02:19:07.598388",
	"_context_user_domain": null,
	"_context_user_name": "admin",
	"publisher_id": "network.an-ocata-control.localdomain",
	"message_id": "d9ed487a-f8f8-4878-b4b6-9a657bdcd9a7"
}
```

### Environment
- OSC Version/Build: 1.0.0-SNAPSHOT-56-g3debe76
- Cloud Provider Info: Ocata openstack
- Plugin Info: n/a
- Other: n/a

### Status
- [x] Reproduced
- [ ] In Progress
- [x] Validated     =>    Verified in Version/Build:

**Comment by arvindn05** [comment]
Make sure to test it in Newton and its functional there

**Comment by mrajora-1** [comment]
Refer to Bug #479 

Relevant Logs
[securityBroker (3).txt](https://github.com/opensecuritycontroller/osc-core/files/1316070/securityBroker.3.txt)

**Comment by larkinscarvalho** [comment]
The payload for the notification is different for Ocata environment. In case of Ocata, device_owner is empty compared to the previous versions of  OpenStack. In case of Newton, the value we receive for the device_owner is "compute:nova". However, in ![OsPortNotificationListener.java#L136](https://github.com/opensecuritycontroller/osc-core/blob/d3ee26a5a92ab6d62231fdc292a3ad1fdb00c6e6/osc-server/src/main/java/org/osc/core/broker/rest/client/openstack/vmidc/notification/listener/OsPortNotificationListener.java#L136) we check for empty device_owner in case of security group with type subnet and not protect external, which result to condition being true and skipping the SG sync.

**Comment by alin2k** [comment]
Albert verified fix in Build#85, both 1)add VM to subnet in SG 2)attached interface to VM in SG; both trigger OSC SG Sync.
